pkg/cloudprovider/provider/scaleway/provider.go

```diff
@@ -0,0 +1,461 @@
+	if err != nil {
+		return nil, nil, err
+	}
+	c.IPv6, err = p.configVarResolver.GetConfigVarBoolValue(rawConfig.IPv6)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.Tags = rawConfig.Tags
+
+	return &c, &pconfig, err
+}
+
+func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec, error) {
+	return spec, nil
+}
+
+func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
+	c, pc, err := p.getConfig(spec.ProviderSpec)
+	if err != nil {
+		return fmt.Errorf("failed to parse config: %v", err)
+	}
+
+	if !validation.IsAccessKey(c.AccessKey) {
+		return fmt.Errorf("invalid access key format '%s', expected SCWXXXXXXXXXXXXXXXXX format", c.AccessKey)
+	}
+	if !validation.IsSecretKey(c.SecretKey) {
+		return fmt.Errorf("invalid secret key format '%s', expected a UUID: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", c.SecretKey)
+	}
+	if !validation.IsProjectID(c.ProjectID) {
+		return fmt.Errorf("invalid project ID format '%s', expected a UUID: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", c.ProjectID)
+	}
+
+	_, err = scw.ParseZone(c.Zone)
+	if err != nil {
+		return err
+	}
+
+	if c.CommercialType == "" {
+		return errors.New("commercial type is missing")
+	}
+
+	_, err = getImageNameForOS(pc.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("invalid operating system specified %q: %v", pc.OperatingSystem, err)
+	}
+
+	return nil
+}
+
+func (p *provider) Create(machine *v1alpha1.Machine, _ *cloudprovidertypes.ProviderData, userdata string) (cloudInstance.Instance, error) {
+	c, pc, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return nil, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	ctx := context.TODO()
+	api, err := c.getInstanceAPI()
+	if err != nil {
+		return nil, err
+	}
+
+	imageName, err := getImageNameForOS(pc.OperatingSystem)
+	if err != nil {
+		return nil, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, invalid operating system specified %q: %v", pc.OperatingSystem, err),
+		}
+	}
+	createServerRequest := &instance.CreateServerRequest{
+		Image:          imageName,
+		Name:           machine.Spec.Name,
+		CommercialType: c.CommercialType,
+		Tags:           append(c.Tags, string(machine.UID)),
+		EnableIPv6:     c.IPv6,
+	}
+
+	serverResp, err := api.CreateServer(createServerRequest, scw.WithContext(ctx))
+	if err != nil {
+		return nil, scalewayErrToTerminalError(err)
+	}
+
+	err = api.SetServerUserData(&instance.SetServerUserDataRequest{
+		Key:      "cloud-init",
+		ServerID: serverResp.Server.ID,
+		Content:  strings.NewReader(userdata),
+	})
+	if err != nil {
+		return nil, scalewayErrToTerminalError(err)
+	}
+
+	klog.V(6).Infof("Scaleway server (id='%s') got fully created", serverResp.Server.ID)
+
+	_, err = api.ServerAction(&instance.ServerActionRequest{
```

Comment:
As we power on the instance on the get method, this actually not needed anymore.